### PR TITLE
Host modal button & spinner order

### LINKED ISSUE #13873: dashboard: Unknown host key modal needs button order and style updates
The key confirmation modal, after adding a new host, needs updating.
 
Screenshot_2020-04-08 Dashboard - localhost localdomain(1)

1. Connect should be on the left.
2. Close should probably be "Cancel" and be a link-styled button
3. Spinner is on the wrong side now (it should be aligned to the right of the modal now)

### PR DESCRIPTION
**Pull request by garrett**
Fixes #13873

### CODE REVIEW

File: pkg/lib/machine-unknown-hostkey.html [hunk 1]
```diff
@@ -16,9 +16,9 @@ <h4 class="modal-title" translate="yes">Unknown Host Key</h4>
 </div>
 
 <div class="modal-footer">
-    <button class="pf-c-button pf-m-secondary" data-dismiss="modal" translate="yes">Close</button>
-
     {{#key}}
         <button class="pf-c-button pf-m-primary" translate="yes">Connect</button>
     {{/key}}
+
+    <button class="pf-c-button pf-m-secondary" data-dismiss="modal" translate="yes">Close</button>
```
Comment (KKoukiou) [inline pkg/lib/machine-unknown-hostkey.html:23]: Do we also maybe want to rename the buttons to close and make it a link like in all other dialogs?

Comment (garrett) [reply pkg/lib/machine-unknown-hostkey.html:23]: Good catch. Yeah, I guess we probably should.

**Comment by garrett** [comment]
Squashed, rebased, included close button → cancel link-styled button.

**Comment by marusak** [comment]
This looks good. I also went through all other pages to find if we have this pattern anywhere else, and here are some finds. Of course this PR does not need to fix all of them, but then we should not fix #13873 with this PR (I think that at least the selinux one should be addressed here as it is the only one remaining that has  the obvious wrong pattern).

- Has only `Close` primary/secondary button without any other action. Not sure if that is right or not. Should we user link-like close buttons if there is no other action? (We use link-like cancel buttons on some other places when it is the only button).
  - pkg/users/index.html
  - pkg/systemd/index.html
  - pkg/shell/index.html (3 times)
  - pkg/networkmanager/index.html
  - pkg/lib/machine-not-supported.html
  - pkg/lib/machine-invalid-hostkey.html
  - pkg/lib/machine-change-port.html
  - pkg/lib/machine-auth-failed.html
  - pkg/docker/index.html

- Sosrepot uses different pattern (pkg/sosreport/index.js) (unrelated, just found it):
differentpattern

- Selinux has it wrong (pkg/lib/cockpit-components-modifications.jsx):
selinuxwrong

**Comment by garrett** [comment]
> Should we user link-like close buttons if there is no other action?

No. Buttons that are link-styled should only be secondary and should be "cancel". Close is an action and should be a button.

Cancel is special. It's basically an anti-action.

Therefore: All of the dialog buttons should look like buttons... except cancel.

More from https://www.patternfly.org/v4/design-guidelines/usage-and-behavior/buttons-and-links#usage-guidelines:
> You might use a link button instead of a secondary button format to create greater visual hierarchy between two buttons. Examples include using text buttons as cancel buttons in modals or wizards, as well as for secondary actions in empty states. 

> Always use link buttons in the following cases:
> - For cancel buttons in wizards or modals
> - For more actions in empty states
> - For action buttons in tables
> - For action buttons in alerts

(Asides: We do use a button for the differently styled "alert" of being in limited access mode, but that's to draw attention to the action, and we basically override the style of the alert anyway, so it's not really an alert in the normal sense. In normal alerts, we should use links for actions. 

Also, we do use buttons in tables for actions, because we do also use links in tables to go places... plus, our tables are from pre-PF3 era and have been upgraded a few times already, so we _could_ reconsider the style of the inline buttons.)

**Comment by garrett** [comment]
I think we should handle the diagnostic report in another PR. The rest of the examples are fine. We do want a close button in those cases, and it should be styled like a button, as it's performing an action of closing the window. 

Canceling is backing out of a dialog, which also happens to close it. Cancelling is basically NOPE NOPE NOPEing out of a place that would otherwise ask you to do something.

Close | Cancel
---|---
grandpa | homer-bushes

**Comment by garrett** [comment]
> Ah, ok. Thought that the selinux one was not, but sure :)

The SELinux dialog is special. The main action isn't really something that would close the dialog. And you're not cancelling a default action. Both could be considered top-level actions, as the copy is optional.

I also wondered about it at first... there's no "primary" but two secondaries. And, yes, that's probably most correct in this one case.

**Pull request merged**